data/data-pipeline/data_pipeline/etl/sources/national_risk_index/etl.py

```diff
@@ -27,8 +25,16 @@ class NationalRiskIndexETL(ExtractTransformLoad):
     AGRIVALUE_LOWER_BOUND = 408000
 
     def __init__(self):
-        self.INPUT_CSV = self.get_tmp_path() / "NRI_Table_CensusTracts.csv"
+        # load YAML config
+        self.DATASET_CONFIG = super().yaml_config_load()
```

Comment:
I tried but you can't define an `__init__` method on the base class, but if you know a way to do it, I would totally go for it.

Reply:
We could place the method on the runner, let me try that.

Reply:
So I spoke too soon. It seems it's hard to do this one, since the runner instance doesn't seem to have the scope to load the class variables from the base `yaml_config_load` and a multitude of tests failed. Unless you can think of something else, I don't know how to tackle this one.

Reply:
Hmm, yeah, this is interesting. I don't have an immediate solution. The scopes are a little funny. Maybe you and I can pair on it, or you can pair with another engineer? To be honest with scope issues like this, when I'm doing it alone I tend to throw spaghetti at the wall, but talking it out with someone can be super helpful.

Reply:
There are three option I see ---

* Do it in  `ExtractTransformLoad.__init__` and then children need to call `super().__init__()` --- not beautiful but also not uncommon. Given that `ExtractTransformLoad` doesn't have an init right now and there's not much else to put in there, this doesn't feel like a net win, unless there's other common setup code that could be extracted.
* Use a metaclass --- pretty sure you could write `yaml_config_load` in terms of a metaclass's `__init__`. Then, you set the metaclass of `ExtractTransformLoad` to your YAML-loading metaclass and children get that functionality. Downside is future maintainability, since like most programmers are not using metaclasses regularly, which can make it hard to reason about them quickly. Upside is what `yaml_config_load` does --- inspecting an external configuration to reflect data there back into the object --- is pretty classic metaclass work and the one place people won't be like "why did you do this?!"
* Write a class decorator that does it --- I think this would be my chosen approach. So extract `yaml_config_load` from `ExtractTransformLoad` and make it a class decorator (which might require a tiny bit of rewriting but not much), then decorate the base `ExtractTransformLoad` with the new yaml_config_load decorator. Children will inherit this decoration and you save the boiler plate but don't fall into madness writing a metaclass.

Finally, tangentially related, not sure you need to call `super()` to call `yaml_config_load` there, since `yaml_config_load` should be inherited like any other member and thereby available on `self`?

Reply:
okay ignore all of what I wrote --- I went to do an example of the decorator approach and learned about `__init_subclass__` and man I guess things have really changed since python 2.4. Anyway if you want to see what it would look like, see this commit: https://github.com/usds/justice40-tool/commit/39229c52fbb7662e94a3e88b05c483d2be020aab

Reply:
What's the resolutions of this - are we comfortable with the code as is?

Reply:
I went ahead and just refactored it in my work on #1518 anyway for fun, so I'm fine with this staying here and you all can decide if you like the `__init_subclass__` approach when you get to that one.

Reply:
I think this is brilliant @mattbowen-usds, had never heard about `__init__subclass__` dunder  -- I'll leave as is for this PR and can get updated when we merge yours.